eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala

```diff
@@ -294,6 +294,13 @@ case class Commitments(channelId: ByteVector32,
 
 object Commitments {
 
+  def validateSeed(commitments: Commitments, keyManager: ChannelKeyManager): Unit = {
+    val localFundingKey = keyManager.fundingPublicKey(commitments.localParams.fundingKeyPath).publicKey
+    val remoteFundingKey = commitments.remoteParams.fundingPubKey
+    val fundingScript = Script.write(Scripts.multiSig2of2(localFundingKey, remoteFundingKey))
+    require(commitments.commitInput.redeemScript == fundingScript, s"public key does not match channel funding transaction for channel ${commitments.channelId}")
```

Comment:
I'm not a fan of throwing here, we have mostly pure functions in this file, can't you return a `Boolean` and let the caller throw / stop eclair?

Reply:
We could but I did not want to give the caller a choice and possibly ignore the return value i.e. it really is an "exception" not an "error" (I even considered calling `sys.exit(1)` here)

Reply:
It's true that it's a fatal error...then do you mind putting that code inside the checker closer to startup instead of inside the `Commitments` object? It shouldn't be used for normal commitments handling

Reply:
since you both agree :) done in https://github.com/ACINQ/eclair/pull/2226/commits/3ad382a40ebe86dbecaa5edf19aab03eb81e96ee (actually I find just returning a boolean more acceptable now that the check has been moved to `DBChecker`